### ISSUE TITLE
consul_z1/0 is failing after update [migrated from diego-release]

### ISSUE THREAD

**Issue by ematpl** [cloudfoundry/cf-release#1102]
> Original issue: https://github.com/cloudfoundry/diego-release/issues/222
> Original poster: @HellenWang

I am deploying CF and Diego to BOSH-Lite follow the guide https://github.com/cloudfoundry/diego-release/tree/develop/examples/bosh-lite.
When I run bosh -n deploy for cf release, it failed with error "Error 400007: `consul_z1/0' is not running after update"

************************
```
D, [2016-11-14 08:32:48 #29393] [] DEBUG -- DirectorJobRunner: Thread is no longer needed, cleaning up
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: Shutting down pool
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: (0.000549s) SELECT "stemcells".* FROM "stemcells" INNER JOIN "deployments_stemcells" ON (("deployments_stemcells"."stemcell_id" = "stemcells"."id") AND ("deployments_stemcells"."deployment_id" = 1))
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: Deleting lock: lock:deployment:cf-warden
D, [2016-11-14 08:32:48 #29393] [] DEBUG -- DirectorJobRunner: Lock renewal thread exiting
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: Deleted lock: lock:deployment:cf-warden
I, [2016-11-14 08:32:48 #29393] [task:22]  INFO -- DirectorJobRunner: sending update deployment error event
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: SENT: hm.director.alert {"id":"a670c3c4-a108-4012-a517-e9aa5e097eb7","severity":3,"title":"director - error during update deployment","summary":"Error during update deployment for cf-warden against Director 71d36859-4f21-446f-8a02-f18d7f1263c6: #<Bosh::Director::AgentJobNotRunning: `consul_z1/0' is not running after update>","created_at":1479112368}
E, [2016-11-14 08:32:48 #29393] [task:22] ERROR -- DirectorJobRunner: `consul_z1/0' is not running after update
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/instance_updater.rb:85:in `update'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/job_updater.rb:74:in `block (2 levels) in update_canary_instance'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh_common-1.2922.0/lib/common/thread_formatter.rb:49:in `with_thread_name'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/job_updater.rb:72:in `block in update_canary_instance'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/event_log.rb:97:in `call'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/event_log.rb:97:in `advance_and_track'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/job_updater.rb:71:in `update_canary_instance'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh-director-1.2922.0/lib/bosh/director/job_updater.rb:65:in `block (2 levels) in update_canaries'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh_common-1.2922.0/lib/common/thread_pool.rb:77:in `call'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh_common-1.2922.0/lib/common/thread_pool.rb:77:in `block (2 levels) in create_thread'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh_common-1.2922.0/lib/common/thread_pool.rb:63:in `loop'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/bosh_common-1.2922.0/lib/common/thread_pool.rb:63:in `block in create_thread'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `call'
/var/vcap/packages/director/gem_home/ruby/2.1.0/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `block in create_with_logging_context'
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: (0.000210s) BEGIN
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: (0.000592s) UPDATE "tasks" SET "state" = 'error', "timestamp" = '2016-11-14 08:32:48.898879+0000', "description" = 'create deployment', "result" = '`consul_z1/0'' is not running after update', "output" = '/var/vcap/store/director/tasks/22', "checkpoint_time" = '2016-11-14 08:32:45.557332+0000', "type" = 'update_deployment', "username" = 'admin' WHERE ("id" = 22)
D, [2016-11-14 08:32:48 #29393] [task:22] DEBUG -- DirectorJobRunner: (0.000569s) COMMIT
I, [2016-11-14 08:32:48 #29393] []  INFO -- DirectorJobRunner: Task took 4 minutes 3.705484818000002 seconds to process.
```
************************

I run bosh logs and found the log is 

************************
```
goroutine 1 [running]:
panic(0x98c600, 0xc4201d16d0)
	/var/vcap/data/packages/golang1.7/48ce7b3927e6f7e6576fff21a7dc9d22b7ba9c62.1-b9e067714c0c471b47996f77d35bc1eadbc8e7ad/src/runtime/panic.go:500 +0x1a1
main.main()
	/var/vcap/data/compile/metron_agent/loggregator/src/metron/main.go:82 +0x8dd
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:09 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:24 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:39 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
2016/11/11 03:52:54 RuntimeStats: failed to emit: EventWriter: No envelope writer set (see SetWriter)
panic: Could not initialize doppler connection pool: Failed to connect to etcd after 15 tries.
************************
```

Could someone please help to check this issue?

**Comment by HellenWang** [comment]
@ematpl can you give me some suggestion to investigate this issue?

**Comment by ematpl** [comment]
@HellenWang I haven't seen this particular error before, so I can't immediately tell you what's wrong. It would be helpful for you to provide some more specifics about both your deployment and this failure:

- What version of CF did you deploy? If you built a dev release, which commit was it based on?
- Did you make sure to generate the BOSH-Lite deployment manifest from that same commit?
- What services are actually failing on the `consul_z1/0` instances? If you run `bosh ssh consul_z1 0` to log into the VM, then run `sudo /var/vcap/bosh/bin/monit summary`, the monit output should tell you. Examining the logs from those services that are failing may indicate the problem.

I suspect that the metron agent panics are spurious: it's complaining because it can't contact the CF etcd cluster, but every new deployment of CF would have the same problem (as it also starts the metron agent on the consul VMs before starting the CF etcd cluster). @wfernandes or someone else from the Loggregator team may be able to provide more insight, though.

Thanks,
Eric

**Comment by HellenWang** [comment]
@ematpl For your question.

1. I deploy CF on the branch "release-candidate". I don't know the concrete version is. I run "cf -v" and print "cf version 6.12.1-56792aa-2015-07-06T22:48:09+00:00"
2. I confirm that I generated the BOSH-Lite deployment manifest from that same commit.
3. I run monit summary on the consul_z1 vm and print 
The Monit daemon 5.2.5 uptime: 4h 34m

Process 'consul_agent'              not monitored - monitor pending
Process 'metron_agent'              Does not exist
System 'system_localhost'           running

I check the log but find not valuable information. I think if it is the Security Configuration for Consul issue. I tried to generate cert through `./scripts/generate-consul-certs`. But it does not work.
Any advices? Thank you very much!

**Comment by dsabeti** [comment]
@HellenWang We'll need a little more information. In both cases below, **please make sure that there isn't any sensitive information such as passwords or private keys**:
1. Please upload the logs for the consul job. They live in `/var/vcap/sys/log/consul/*`. This should help us figure out what's going wrong.
2. Please upload the deployment manifest that you used.

Thanks,
David Sabeti
PM Release Integration Team

**Comment by dsabeti** [comment]
@HellenWang, do you still need help with this? I'm planning to close this issue on Monday otherwise.

**Comment by HellenWang** [comment]
@dsabeti Sorry for delay.
The consul_agent.zip is the package of the logs for the consul job.
[consul_agent.zip](https://github.com/cloudfoundry/cf-release/files/630341/consul_agent.zip)
cf.txt is the manifest that I used.
[cf.txt](https://github.com/cloudfoundry/cf-release/files/630354/cf.txt)

**Comment by ChunyiLyu** [comment]
@HellenWang Sorry for the late response. Looks like metron_agent is also having trouble starting up. If you are still having this issue, would you mind sharing the logs for metron_agent job? They will be in ` /var/vcap/sys/log/metron_agent/* `

Have you try to redeploy? Did the deployment keep failing the same way each time?

**Comment by ChunyiLyu** [comment]
@HellenWang We are closing this due to inactivity.  Please feel free to reopen the issue if you are still having trouble.
